src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -1571,6 +1571,128 @@ void C2_MacroAssembler::vinsert(BasicType typ, XMMRegister dst, XMMRegister src,
   }
 }
 
+#ifdef _LP64
+void C2_MacroAssembler::vgather8b_masked_offset(BasicType elem_bt,
+                                                XMMRegister dst, Register base,
+                                                Register idx_base,
+                                                Register offset, Register mask,
+                                                Register mask_idx, Register rtmp,
+                                                int vlen_enc) {
+  vpxor(dst, dst, dst, vlen_enc);
+  if (elem_bt == T_SHORT) {
+    Label case0, case1, case2, case3;
+    Label* larr[] = {&case0, &case1, &case2, &case3};
```

Comment:
Not sure if I asked this already: why define them all here, rather than locally in the loop?

Reply:
To avoid invariant initializations to happen within the loop, compiler will unroll this small loop and will forward the initializations, if it does not then we can save redundant allocation within loop.

Reply:
I could not find any other case with the same pattern, of initializing a list of Labels.

On the other hand, I can find cases where we already do what I am saying:
`C2_MacroAssembler::rtm_counters_update`

Reply:
Hi @eme64 , I was referring to allocation of label's array.  To be concise and avoid hand unrolling of loop, I chose an array of labels.

Reply:
I was referring to the various arrays as well above. I think it would be exactly more concise if you defined a local label in the loop body.

Reply:
Correct, with each successive iteration label will be bound to a new offset within code blob, so moving it within loop is a good suggestion.